packages/analytics/amplify_analytics_pinpoint_dart/lib/src/impl/analytics_client/event_client/event_client.dart

```diff
@@ -215,21 +229,26 @@ class EventClient implements Closeable {
   }
 
   // If exception is recoverable, do not delete eventIds from local cache
-  void _handleRecoverableException(
-    SmithyHttpException e,
-    Map<String, StoredEvent> eventIdsToDelete,
-  ) {
+  void _logRecoverableException(Exception e) {
```

Comment:
Why log a recoverable exception as a warning? This would be expected if a client goes offline.

Imagine a customer is streaming all warning / error logs to a remote server. If a client goes offline for a while, their logs will get filled with these warnings.

Reply:
I don't think any changes were made around this. Is there a good reason to have these as a warning? Recoverable issues that will be retried don't seem like warnings to me. They are expected.

Reply:
Maybe log it as a warning after it hits the retry limit?

Reply:
Hi Jordan I updated the code to only log recoverable exceptions for non network offline exceptions.  So the logs should not be filled with these warnings. 

I don't think they are expected, though that doesn't mean we have to log it though.  

The underlying error received is a Pinpoint `InternalServerErrorException` due to an 'unknown server error'.  

I guess we could argue since the user probably can't fix an Pinpoint internal error and that events will be retried anyway that this logging isn't necessary. 

Do you know if we have an Amplify specific concept of what a warning should be?  Maybe it should be logged still but at the info level as this is not expected, but still recoverable. 

And yes we already log if retry limit is hit.

Reply:
If the network failures are not being logged, then this is fine. My main concern was what happened when a user went offline.